tests/test_headers.py

```diff
@@ -117,6 +117,16 @@ def on_head(self, req, resp):
         resp.content_location = self.URL1
 
 
+class UnicodeHeaderResource:
+
+    def on_get(self, req, resp):
+        resp.set_headers([
+            (u'X-auTH-toKEN', 'toomanysecrets'),
+            ('Content-TYpE', u'application/json'),
+            (u'X-symBOl', u'\u0040'),
```

Comment:
@ginoledesma  OK, on second thought, this test exposes an issue that is different from #413. I don't think we should try to solve for detecting what types of characters are in the string, because (a) most HTTP header fields only use a subset of US-ASCII and (b) it would require additional run-time cost to do so.

> Historically, HTTP has allowed field content with text in the ISO-8859-1 charset [ISO-8859-1], supporting other charsets only through use of [RFC2047] encoding.  In practice, most HTTP header field values use only a subset of the US-ASCII charset [USASCII]. Newly defined header fields SHOULD limit their field values to US-ASCII octets.  A recipient SHOULD treat other octets in field content (obs-text) as opaque data.

https://tools.ietf.org/html/rfc7230#section-3.2.4

So anyway, let's not add that negative test after all. If we need to add stronger checking later, we can always do that.